### PR TITLE
add compat for dockerfile compatability newrelic-nri-statsd

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: newrelic-nri-statsd.yaml [hunk 1]
```diff
@@ -32,6 +32,14 @@ pipeline:
       install -Dm755 nri-statsd.sh "${{targets.destdir}}"/home/nonroot/nri-statsd.sh
       install -Dm755 run-statsd.sh "${{targets.destdir}}"/home/nonroot/run-statsd.sh
 
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Make the nri-statsd.sh script compatible with Dockerfile entrypoint
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"
+          ln -sf /home/nonroot/nri-statsd.sh "${{targets.subpkgdir}}"/nri-statsd.sh
```
Comment (EyeCantCU) [inline newrelic-nri-statsd.yaml:41]: Can this be avoided by explicitly setting the entry point as `/home/nonroot/nri-statsd.sh`?

Comment (developer-guy) [reply newrelic-nri-statsd.yaml:41]: its working when I set `./nri-statsd.sh` not when its `/nri-statsd.sh` 🤷🏻‍♂️

Comment (developer-guy) [reply newrelic-nri-statsd.yaml:41]: but good suggestion, let me try that one!

Comment (developer-guy) [reply newrelic-nri-statsd.yaml:41]: it sounds like its worked! 

```
(っ◔◡◔)っ  docker container run --pull always --rm -p 8125:8125 ghcr.io/developer-guy/newrelic-nri-statsd:latest                                             1.22.2
WARNING: configuration file not found, generating one based on default values.
make sure you have provided the required NR_API_KEY and optionally
NR_ACCOUNT_ID if using Insights events, via environment variables.
time="2024-04-07T14:09:31Z" level=info msg="Starting server"
time="2024-04-07T14:09:31Z" level=info msg="No cloud provider specified"
time="2024-04-07T14:09:31Z" level=info msg="internal metrics OFF, to enable set 'statser-type' to 'logging' or 'internal'" backend=newrelic
time="2024-04-07T14:09:31Z" level=fatal msg="could not init backend \"newrelic\": [newrelic] api-key is required to flush to insights & metrics backends"
```


```
(っ◔◡◔)っ  crane config --platform linux/arm64 ghcr.io/developer-guy/newrelic-nri-statsd:latest | jq '.config' 
{
  "Entrypoint": [
    "/home/nonroot/nri-statsd.sh"
  ],
  "Env": [
    "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
    "SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt"
  ],
  "User": "65532",
  "WorkingDir": "/home/nonroot"
}
```

Comment (developer-guy) [reply newrelic-nri-statsd.yaml:41]: thanks @EyeCantCU !!!

**Pull request closed without merging**